### ISSUE TITLE
no suitable method found for exerciseScheduleAppointment(java.time.LocalDate)

### ISSUE THREAD

**Issue by runeksvendsen** [obsidiansystems/ex-healthcare-claims-processing#65]
> > For `integration_test` I'm seeing the following error at the "Integration tests"-step (command: `mvn verify`), which I don't know how to solve:
> > 
> > ```
> > [ERROR] /home/circleci/project/src/test/java/com/daml/product/healthcareclaims/HealthcareClaimsProcessingIT.java:[116,47] no suitable method found for exerciseScheduleAppointment(java.time.LocalDate)
> >     method com.digitalasset.nanobot.healthcare.models.main.provider.ReferralDetails.ContractId.exerciseScheduleAppointment(com.digitalasset.nanobot.healthcare.models.main.provider.ScheduleAppointment) is not applicable
> >       (argument mismatch; java.time.LocalDate cannot be converted to com.digitalasset.nanobot.healthcare.models.main.provider.ScheduleAppointment)
> >     method com.digitalasset.nanobot.healthcare.models.main.provider.ReferralDetails.ContractId.exerciseScheduleAppointment(java.time.Instant) is not applicable
> >       (argument mismatch; java.time.LocalDate cannot be converted to java.time.Instant)
> > ```
> > 
> > @richardkapolnai-da would you happen to know what's going on here?
> 
> Probably Java code was not updated when the appointments were changed from a Date to a Time in the Daml model. Should be an easy fix in Java.

_Originally posted by @richardkapolnai-da in https://github.com/obsidiansystems/ex-healthcare-claims-processing/pull/58#issuecomment-848699567_

**Comment by runeksvendsen** [comment]
@richardkapolnai-da I'm not familiar with the Java code, so if it's an easy fix for you would you mind taking a stab at solving this?

Here's [the full error I'm seeing in CI](https://app.circleci.com/pipelines/github/obsidiansystems/ex-healthcare-claims-processing/131/workflows/47c9c0de-db01-4a59-9529-acd912b3ab42/jobs/316/parallel-runs/0/steps/0-106):

```
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/circleci/project/src/test/java/com/daml/product/healthcareclaims/HealthcareClaimsProcessingIT.java:[116,47] no suitable method found for exerciseScheduleAppointment(java.time.LocalDate)
    method com.digitalasset.nanobot.healthcare.models.main.provider.ReferralDetails.ContractId.exerciseScheduleAppointment(com.digitalasset.nanobot.healthcare.models.main.provider.ScheduleAppointment) is not applicable
      (argument mismatch; java.time.LocalDate cannot be converted to com.digitalasset.nanobot.healthcare.models.main.provider.ScheduleAppointment)
    method com.digitalasset.nanobot.healthcare.models.main.provider.ReferralDetails.ContractId.exerciseScheduleAppointment(java.time.Instant) is not applicable
      (argument mismatch; java.time.LocalDate cannot be converted to java.time.Instant)
[ERROR] /home/circleci/project/src/test/java/com/daml/product/healthcareclaims/HealthcareClaimsProcessingIT.java:[186,47] no suitable method found for exerciseScheduleAppointment(java.time.LocalDate)
    method com.digitalasset.nanobot.healthcare.models.main.provider.ReferralDetails.ContractId.exerciseScheduleAppointment(com.digitalasset.nanobot.healthcare.models.main.provider.ScheduleAppointment) is not applicable
      (argument mismatch; java.time.LocalDate cannot be converted to com.digitalasset.nanobot.healthcare.models.main.provider.ScheduleAppointment)
    method com.digitalasset.nanobot.healthcare.models.main.provider.ReferralDetails.ContractId.exerciseScheduleAppointment(java.time.Instant) is not applicable
      (argument mismatch; java.time.LocalDate cannot be converted to java.time.Instant)
[INFO] 2 errors 
```

**Comment by runeksvendsen** [comment]
Fixed by #73
